OfflineAudioContext/explainer.md

```diff
@@ -0,0 +1,232 @@
+
+**Cons**
+
+- requires the output of `startStreamingRendering()` to return an array of `Float32Array` in planar format for each output channel
+- this leaves a question of what to do if only one channel is read by the consumer, i.e. what should happen to the other channel's data?
+
+#### `f32-interleaved`
+
+**Pros**
+
+- allows for streaming a single stream of data, rather than one for each channel
+- interoperates with WebCodecs APIs because they operate with interleaved streams of data
+
+**Cons**
+
+- introduces a new type to the WebAudio spec, `f32-interleave` which does not exist at the moment
+
+#### Recommendation
+
+`f32-interleaved` as it is the most interoperable with other media APIs, like WebCodecs, and simplifies processing with other data streams such as video.
+
+## Alternative 1 - Modify existing `startRendering` method to allow streaming output
+
+An alternative approach is to add options to the existing `startRendering()` to configure its operating mode. The mode can be set to `stream` to achieve streaming output. This is similar to the proposed approach but rather than adding a new function, it re-uses an existing function.
+
+```js
+interface OfflineAudioRenderingOptions {
+    mode: "audiobuffer" | "stream"
+}
+
+interface OfflineAudioContext {
+    Promise<AudioBuffer | ReadableStream> startRendering(optional: startRenderingOptions);
+}
+```
+
+Usage example:
+
+```js
+const offlineContext = new OfflineAudioContext(...);
+
+// Add some nodes to build a graph...
+
+const reader = await offlineContext.startRendering(options: { mode: "stream"}).reader();
+while (true) {
+    // get the next chunk of data from the stream
+    const result = await reader.read();
+
+    // the reader returns done = true when there are no more chunks to consume
+    if (result.done) {
+        break;
+    }
+
+    const buffers = result.value;
+}
+```
+
+The existing API remains unchanged for backwards compatibility:
+
+```js
+/**
+ * Existing API unchanged
+ */
+const offlineContext = new OfflineAudioContext(...);
+
+// Add some nodes to build a graph...
+
+// Full AudioBuffer is allocated
+const renderedBuffer = await offlineContext.startRendering();
+```
+
+### Pros
+
+- The same pros as the proposed approach
+
+### Cons
+
+- The same cons at the proposed approach
+- Less explicit than the proposed approach as it overloads an existing public API function. It is safer and simpler to add a new function and not change the behaviour of an existing function
+
+## Alternative 2 - emit `ondataavailable` events
+
+Keep current `startRendering()` API but do not allocate the full `AudioBuffer`. After starting, periodically emit events on the context or a new interface such as `ondataavailable(chunk: AudioBuffer)`.
+
+The user can subscribe and collect chunks for processing.
+
+At the end, the API may optionally still provide a full `AudioBuffer`.
+
+### Pros
+
+- Simple to integrate with existing event-driven patterns
+
+### Cons
```

Comment:
I think the main con is that the web platform already has a pattern for streaming (ReadableStreams).  Using this provides interop with the rest of the platform that uses ReadableStreams.  You implicitly mentioned this above with the BYOB reader.  ReadableStreams also provides a pattern that many developers are familiar with.  One other potential alternative would be to use a MediaStream instead of a ReadableStream.  The non-offline AudioContext provides MediaStreams through the MediaStreamAudioDestinationNode.